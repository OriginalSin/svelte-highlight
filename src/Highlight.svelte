<script>
  import { hljs } from './hljs';
  export let language = { name: undefined, register: undefined };
  export let className = 'svelte-highlight';

  let block;

  $: {
    if (language.name && language.register) {
      hljs.registerLanguage(language.name, language.register);
    }

    if (block) {
      hljs.highlightBlock(block);
    }
  }
</script>

<pre bind:this={block} class={className}>
  <code>
    <slot />
  </code>
</pre>
