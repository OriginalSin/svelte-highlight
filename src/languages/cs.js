import cs from 'highlight.js/lib/languages/cs';

export default { name: 'cs', register: cs };
