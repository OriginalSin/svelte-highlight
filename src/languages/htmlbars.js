import htmlbars from 'highlight.js/lib/languages/htmlbars';

export default { name: 'htmlbars', register: htmlbars };
