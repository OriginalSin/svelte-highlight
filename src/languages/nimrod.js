import nimrod from 'highlight.js/lib/languages/nimrod';

export default { name: 'nimrod', register: nimrod };
