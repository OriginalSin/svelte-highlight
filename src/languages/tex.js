import tex from 'highlight.js/lib/languages/tex';

export default { name: 'tex', register: tex };
