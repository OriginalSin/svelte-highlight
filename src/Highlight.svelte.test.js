/* eslint-disable no-underscore-dangle */
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/svelte';
import Highlight from './Highlight.wrapper.svelte';
import { typescript } from './languages';

it('matches the snapshot', () => {
  const props = {
    className: 'svelte-highlight',
    language: typescript,
    _code: 'const a: number = 4;'
  };
  const { container, component } = render(Highlight, { props });

  expect(component).toMatchSnapshot();
  expect(container.firstChild).toHaveTextContent(props._code);
});
