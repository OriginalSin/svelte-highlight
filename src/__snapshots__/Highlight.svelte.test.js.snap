// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
Highlight_wrapper {
  "$$": Object {
    "after_update": Array [],
    "before_update": Array [],
    "bound": Object {},
    "callbacks": Object {},
    "context": Map {},
    "ctx": Object {
      "_code": "const a: number = 4;",
      "className": "svelte-highlight",
      "language": Object {
        "name": "typescript",
        "register": [Function],
      },
    },
    "dirty": null,
    "fragment": Object {
      "c": [Function],
      "d": [Function],
      "i": [Function],
      "l": [Function],
      "m": [Function],
      "o": [Function],
      "p": [Function],
    },
    "not_equal": [Function],
    "on_destroy": Array [
      [Function],
    ],
    "on_mount": Array [],
    "props": Array [
      "className",
      "language",
      "_code",
    ],
    "update": [Function],
  },
  "$capture_state": [Function],
  "$inject_state": [Function],
  "$set": [Function],
}
`;
